src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -690,8 +690,14 @@ string IDurableOrchestrationContext.StartNewOrchestration(string functionName, o
             {
                 // Check to see if CallHttpAsync() threw a TimeoutException
                 // In this case, we want to throw a TimeoutException instead of a FunctionFailedException
-                if (functionName.Equals(HttpOptions.HttpTaskActivityReservedName) && e.InnerException is TimeoutException)
+                if (functionName.Equals(HttpOptions.HttpTaskActivityReservedName) &&
+                    (e.InnerException is TimeoutException || e.InnerException is HttpRequestException))
                 {
+                    if (e.InnerException is HttpRequestException)
+                    {
+                        throw new HttpRequestException(e.Message);
```

Comment:
I tried to follow the same pattern as the `TimeoutException` but the `Message` property in `HttpRequestException` doesn't serialize the same way and the value doesn't get passed. That's why I chose to create a new `HttpRequestException` in this method instead of in `RunAsync()` in TaskHttpActivityShim.cs.

Reply:
Interesting! Can you clarify how is the serialization different?

Reply:
ah, I see you clarify this in a later comment. Nevermind